driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedCrudHelper.java

```diff
@@ -0,0 +1,627 @@
+            }
+        }
+        try {
+            return OperationResult.of(new BsonArray(iterable.into(new ArrayList<>())));
+        } catch (Exception e) {
+            return OperationResult.of(e);
+        }
+    }
+
+    OperationResult executeFindOneAndUpdate(final BsonDocument operation) {
+        MongoCollection<BsonDocument> collection = entities.getCollection(operation.getString("object").getValue());
+        BsonDocument arguments = operation.getDocument("arguments");
+
+        BsonDocument filter = null;
+        BsonDocument update = null;
+        FindOneAndUpdateOptions options = new FindOneAndUpdateOptions();
+
+        for (Map.Entry<String, BsonValue> cur : arguments.entrySet()) {
+            switch (cur.getKey()) {
+                case "filter":
+                    filter = cur.getValue().asDocument();
+                    break;
+                case "update":
+                    update = cur.getValue().asDocument();
+                    break;
+                case "returnDocument":
+                    switch (cur.getValue().asString().getValue()) {
+                        case "Before":
+                            options.returnDocument(ReturnDocument.BEFORE);
+                            break;
+                        case "After":
+                            options.returnDocument(ReturnDocument.AFTER);
+                            break;
+                        default:
+                            throw new UnsupportedOperationException("Can't happen");
+                    }
+                    break;
+                default:
+                    throw new UnsupportedOperationException("Unsupported argument: " + cur.getKey());
+            }
+        }
+        requireNonNull(filter);
+        requireNonNull(update);
+
+        try {
+            BsonDocument result = collection.findOneAndUpdate(filter, update, options);
+            return OperationResult.of(result);
+        } catch (Exception e) {
+            return OperationResult.of(e);
+        }
+    }
+
+    OperationResult executeAggregate(final BsonDocument operation) {
+        String entityName = operation.getString("object").getValue();
+
+        BsonDocument arguments = operation.getDocument("arguments");
+        List<BsonDocument> pipeline = arguments.getArray("pipeline").stream().map(BsonValue::asDocument).collect(toList());
+        AggregateIterable<BsonDocument> iterable;
+        if (entities.hasDatabase(entityName)) {
+            iterable = entities.getDatabase(entityName).aggregate(requireNonNull(pipeline), BsonDocument.class);
+        } else if (entities.hasCollection(entityName)) {
+            iterable = entities.getCollection(entityName).aggregate(requireNonNull(pipeline));
+        } else {
+            throw new UnsupportedOperationException("Unsupported entity type with name: " + entityName);
+        }
+        for (Map.Entry<String, BsonValue> cur : arguments.entrySet()) {
```

Comment:
This seems an odd loop & switch - just to check for unsupported keys.

Reply:
I put it in because in the near future there will be options